CMakeLists.txt

```diff
@@ -1,87 +1,23 @@
-cmake_minimum_required(VERSION 3.5)
-
-project(spdlog_vendor)
+cmake_minimum_required(VERSION 3.10)
+project(spdlog_vendor CXX)
 
 find_package(ament_cmake REQUIRED)
-ament_add_default_options()
-
-# Avoid DOWNLOAD_EXTRACT_TIMESTAMP warning for CMake >= 3.24
-if(POLICY CMP0135)
-  cmake_policy(SET CMP0135 NEW)
-endif()
-
-option(FORCE_BUILD_VENDOR_PKG
-  "Build spdlog from source, even if system-installed package is available"
-  OFF)
-
-if(NOT FORCE_BUILD_VENDOR_PKG)
-  find_package(spdlog QUIET)
-endif()
+find_package(ament_cmake_vendor_package REQUIRED)
 
-macro(build_spdlog)
-  set(extra_cmake_args)
-  if(DEFINED CMAKE_BUILD_TYPE)
-    list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
-  endif()
-  if(DEFINED CMAKE_TOOLCHAIN_FILE)
-    list(APPEND extra_cmake_args "-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}")
-    if(ANDROID)
-      if(DEFINED ANDROID_ABI)
-        list(APPEND extra_cmake_args "-DANDROID_ABI=${ANDROID_ABI}")
-      endif()
-      if(DEFINED ANDROID_CPP_FEATURES)
-        list(APPEND extra_cmake_args "-DANDROID_CPP_FEATURES=${ANDROID_CPP_FEATURES}")
-      endif()
-      if(DEFINED ANDROID_FUNCTION_LEVEL_LINKING)
-        list(APPEND extra_cmake_args "-DANDROID_FUNCTION_LEVEL_LINKING=${ANDROID_FUNCTION_LEVEL_LINKING}")
-      endif()
-      if(DEFINED ANDROID_NATIVE_API_LEVEL)
-        list(APPEND extra_cmake_args "-DANDROID_NATIVE_API_LEVEL=${ANDROID_NATIVE_API_LEVEL}")
-      endif()
-      if(DEFINED ANDROID_NDK)
-        list(APPEND extra_cmake_args "-DANDROID_NDK=${ANDROID_NDK}")
-      endif()
-      if(DEFINED ANDROID_STL)
-        list(APPEND extra_cmake_args "-DANDROID_STL=${ANDROID_STL}")
-      endif()
-      if(DEFINED ANDROID_TOOLCHAIN_NAME)
-        list(APPEND extra_cmake_args "-DANDROID_TOOLCHAIN_NAME=${ANDROID_TOOLCHAIN_NAME}")
-      endif()
-    endif()
-  else()
-    list(APPEND extra_cmake_args "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}")
-  endif()
-  list(APPEND extra_cmake_args "-DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}")
-  list(APPEND extra_cmake_args "-DSPDLOG_BUILD_TESTS=OFF")
-  list(APPEND extra_cmake_args "-DSPDLOG_BUILD_EXAMPLE=OFF")
-  if(NOT WIN32 AND BUILD_SHARED_LIBS)
-    list(APPEND extra_cmake_args "-DSPDLOG_BUILD_SHARED=ON")
-  endif()
+find_package(spdlog 1.5.0 QUIET)
 
-  include(ExternalProject)
+# While most CMake projects build shared libraries by default, it appears that
+# spdlog needs to be told to do so explicitly.
+set(BUILD_SHARED_LIBS TRUE CACHE BOOL "Build shared libraries")
```

Comment:
Out of curiousity, why is this set here rather than being passed through to `ament_vendor` as a `CMAKE_ARGS`?

Reply:
`BUILD_SHARED_LIBS` is one of the selected CMake variables that get implicitly passed to the vendor project if defined. It was less code to treat it as a project-local variable than it was to add conditional logic for adding it to the explicit `CMAKE_ARGS` to the `ament_vendor` call.